superset/models/core.py

```diff
@@ -325,16 +325,13 @@ def get_sqla_engine(
             params["poolclass"] = NullPool
 
         connect_args = params.get("connect_args", {})
-        configuration = connect_args.get("configuration", {})
 
-        # If using Hive, this will set hive.server2.proxy.user=$effective_username
-        configuration.update(
-            self.db_engine_spec.get_configuration_for_impersonation(
-                str(sqlalchemy_url), self.impersonate_user, effective_username
-            )
+        # If using presto, this will set principal_username=$effective_username
+        # If using Hive, this will set hive.server2.proxy.user=$effective_username on connect_args['configuration']
```

Comment:
I would move this comment to `hive.py`, as `core.py` should be as generic as possible.